### ISSUE TITLE
Cannot click on conflicts in the first file of a commit

### ISSUE THREAD

**Issue by plichard** [gitbutlerapp/gitbutler#5826]
### Version

0.14.3

### Operating System

Linux

### Distribution Method

AppImage (Linux)

### Describe the issue

Clicking on the first conflicted file in a commit just closes the commit details instead of opening the conflict handling ui.

https://github.com/user-attachments/assets/a4ed85be-6717-46ec-9fe0-c08e00f87813

**Comment by Byron** [comment]
Thanks a lot for reporting!

I noticed this behaviour as well and thought it should show the conflict itself, or inform me how to `Resolve Conflicts`.
Conflict resolution is already possible, but only after the respective button was pressed.

**Comment by plichard** [comment]
Oh! Right, this has nothing to do with it being the first file in the commit listing, for some reason the "Resolve conflicts" button is often just invisible to my brain, you mentioning it revealed it :sweat_smile: .
I forgot conflict resolution worked for me previously because I actually clicked "Resolve conflicts" first, and then I was able to select one of the files (which was not the first one at the time).

This report must have been a confusing read ...

**Comment by Byron** [comment]
> Oh! Right, this has nothing to do with it being the first file in the commit listing, for some reason the "Resolve conflicts" button is often just invisible to my brain, you mentioning it revealed it 😅 .

This is super-interesting! I noticed that too, and magically, after #5722 was done, this issue disappeared for me.
Could you find an occasion where this happens in the latest stable, and then [try a nightly build](https://app.gitbutler.com/downloads)? Maybe that issue goes away.

Then what stays here is the UX issue that people right now are visually trained that something happens once they click a file. Maybe while that's not the case, these conflicted files could be presented differently. CC @PavelLaptev .

**Comment by PavelLaptev** [comment]
@Byron I agree with you and @plichard. We should be able to at least preview the file when we click on it. I'll create a PR.

For the next iteration, we can also think about additional ways to go to the conflict resolution page.

**Comment by Caleb-T-Owens** [comment]
We currently don't load that file data. It might take some work to get a file display working.

We'd need to generate the files with conflicts written in them for display, and to have consider how to display the cases where you've modified and they've deleted

**Comment by PavelLaptev** [comment]
@Caleb-T-Owens got it. I thought it's only a frontend thing, just a svelte condition that prevent to click on a conflicted file row.

**Comment by Caleb-T-Owens** [comment]
I wonder if we could have them do a little red wiggle and have a toast or little message appear, just so the user knows why they are not clickable?

**Comment by PavelLaptev** [comment]
@Caleb-T-Owens yes, let me think about this. But it's a good idea for a quick workaround.

**Comment by Byron** [comment]
My intuition here was that conflicting files should be displayed in a way that makes clear they aren't clickable in the first place, by being visually different enough from normal files. Something I also repeatedly tried to do is to mouse-over the conflict-icon, to learn what that should mean, but nothing happened.

**Comment by PavelLaptev** [comment]
When you click on a conflicted file, what if we display the warning in the same way we show files? The warning text would appear alongside a button to open conflict resolution mode.

<img width="858" alt="image" src="https://github.com/user-attachments/assets/769987ef-beea-4a2b-9052-283f18f7377c" />

We are already doing something similar for large files:

<img width="894" alt="image" src="https://github.com/user-attachments/assets/a5951128-4029-4c0a-8b33-516148131c79" />

**Comment by Byron** [comment]
That would make sense to me, and it appears to be consistent as well with what users expect. The comparison to how large files are handled makes that clear.

**Comment by Caleb-T-Owens** [comment]
I like the look of that

**Comment by PavelLaptev** [comment]
Cool! @Caleb-T-Owens, could you please help me with this? I'm struggling to figure out how to open `FileDiff` for conflicted files. Currently, we open the `FileDiff` if there’s a file in `selectedFileIds`, but `conflictedFiles` are not the same as `displayedFiles`, so I can’t just pass it to `selectFilesInList`.

![image](https://github.com/user-attachments/assets/3968a1e4-4cca-4a31-80fb-98e2c36579c1)

**Comment by krlvi** [comment]
Hey @plichard, this was resolved in https://github.com/gitbutlerapp/gitbutler/pull/9593. Conflicted files are now clickable and prompt you to enter conflict resolution before previewing them.

The fix shipped in [0.15.7](https://github.com/gitbutlerapp/gitbutler/releases/tag/release/0.15.7).
